src/main/scala/sigmastate/serialization/DataJsonEncoder.scala

```diff
@@ -0,0 +1,217 @@
+package sigmastate.serialization
+
+import java.nio.charset.StandardCharsets
+
+import io.circe._
+import io.circe.syntax._
+import sigmastate._
+import sigmastate.eval.SigmaDsl
+import sigmastate.lang.exceptions.SerializerException
+import special.collection.Coll
+import special.sigma.BigInt
+import spire.syntax.all.cfor
+
+import scala.collection.mutable
+import scala.collection.mutable.{MutableList, Stack}
+
+object STypeNameEncoder {
+  def encode[T <: SType](tpe: T): String = tpe match {
+    case SUnit => "Unit"
+    case SBoolean => "Boolean"
+    case SByte => "Byte"
+    case SShort => "Short"
+    case SInt => "Int"
+    case SLong => "Long"
+    case SBigInt => "BigInt"
+    case SString => "String"
+    case tColl: SCollectionType[a] =>
+      "Coll[" + encode(tColl.elemType) + "]"
+    case t => throw new SerializerException(s"Not defined DataJsonEncoder for ${t}")
+  }
+
+  sealed trait Chunk
+
+  case class NameChunk(data: String)
+  case class ParenthesesChunk(chunks: List[Chunk])
+  case class BracketsChunk(typename: NameChunk, wrappedTypeChunk: Chunk)
+
+  private class ChunkParser(from: String) {
```

Comment:
Think low level parsing is not necessary with circe.

Reply:
removed